lxd/storage/drivers/backups.go

```diff
@@ -0,0 +1,100 @@
+package drivers
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/lxc/lxd/lxd/operations"
+	"github.com/lxc/lxd/shared"
+)
+
+// genericBackupUnpack unpacks a non-optimized backup tarball through a storage driver.
+func genericBackupUnpack(d Driver, poolName string, vol Volume, snapshots []string, srcData io.ReadSeeker, op *operations.Operation) (func(vol Volume) error, func(), error) {
+	revert := true
+
+	// Define a revert function that will be used both to revert if an error occurs inside this
+	// function but also return it for use from the calling functions if no error internally.
+	revertHook := func() {
```

Comment:
@stgraber unless I'm missing something, this isn't returned at the end as the revertHook anymore.

Reply:
Ah yeah, it should be. Fixed.